### PR TITLE
Add DatabaseHandler to enable Delete Database operations

### PR DESCRIPTION
**Pull request by corivera**
I wasn't able to get Create Database working perfectly, so I've removed those parts to get this in before code complete this week (Save & Script). I left in the Create Database dialog creation portions though, since those methods are abstract in the base class and also work fine as-is. These changes are needed for this ADS PR: https://github.com/microsoft/azuredatastudio/pull/22948

### CODE REVIEW

File: src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs [hunk 1]
```diff
@@ -0,0 +1,469 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+#nullable disable
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs:6]: As you're continuing to work on this, please see about enabling nullable in the files you're working on and fixing the warnings that show up.

File: src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs [hunk 2]
```diff
@@ -0,0 +1,469 @@
...
+using System.Linq;
+using Microsoft.SqlTools.ServiceLayer.Admin;
+using static Microsoft.SqlTools.ServiceLayer.Admin.AzureSqlDbHelper;
+using System.Globalization;
+using System.Xml;
+
+namespace Microsoft.SqlTools.ServiceLayer.ObjectManagement
+{
+    /// <summary>
+    /// Login object type handler
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs:26]:
```suggestion
    /// Database object management handler
```

File: src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs [hunk 3]
```diff
@@ -0,0 +1,469 @@
...
+    /// </summary>
+    public class DatabaseHandler : ObjectTypeHandler<DatabaseInfo, DatabaseViewContext>
+    {
+        private const int minimumVersionForWritableCollation = 8;
+        private const int minimumVersionForRecoveryModel = 8;
+        private const string DefaultValue = "<default>";
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs:32]: This should be localized, right?

File: src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseInfo.cs [hunk 4]
```diff
@@ -0,0 +1,19 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+namespace Microsoft.SqlTools.ServiceLayer.ObjectManagement
+{
+    /// <summary>
+    /// a class for storing various login properties
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseInfo.cs]:
```suggestion
    /// Stores various database properties
```

File: src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs [hunk 5]
```diff
@@ -0,0 +1,469 @@
...
+
+        public override bool CanHandleType(SqlObjectType objectType)
+        {
+            return objectType == SqlObjectType.Database;
+        }
+
+        public override Task<InitializeViewResult> InitializeObjectView(InitializeViewRequestParams requestParams)
+        {
+            // open a connection for running the database dialog and associated task
+            ConnectionInfo originalConnInfo;
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs:63]: There's a lot of code in here, in the interest of being able to accurately review it in a timely manner is there any way you could just temporarily delete the stuff you don't need? I would rather avoid shoving a bunch of unreviewed stuff in here, even if it isn't actively being used.

File: src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseViewContext.cs [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+            this.Connection = connection;
+        }
+
+        public ServerConnection Connection { get; }
+
+        public override void Dispose()
+        {
+            try
+            {
+                this.Connection.Disconnect();
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseViewContext.cs:24]: This connection is being passed in from the data container - shouldn't that be the thing that does the cleanup?

File: src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseViewInfo.cs [hunk 7]
```diff
@@ -0,0 +1,18 @@
+
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+#nullable disable
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseViewInfo.cs:6]: This should be easy enough to enable nullable stuff for - probably just initialize them to an empty array right?

File: src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseViewInfo.cs [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+namespace Microsoft.SqlTools.ServiceLayer.ObjectManagement
+{
+    /// <summary>
+    /// A class for storing various properties needed for initializing New Database dialog
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseViewInfo.cs:11]: Why is this only for a new database? What about viewing properties? Or is new objects the only thing that these "view info" classes are for?

@alanrenmsft This is what I was mentioning to you, having clear documentation around what the base classes for these things are supposed to be used for would be great.

File: src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs (hunk 5 above)
Comment (corivera) [reply src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs]: Removed the extra stuff

File: src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseViewContext.cs (hunk 6 above)
Comment (corivera) [reply src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseViewContext.cs]: Removed

File: src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseViewInfo.cs (hunk 8 above)
Comment (corivera) [reply src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseViewInfo.cs]: Removed for now

File: src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs (hunk 3 above)
Comment (corivera) [reply src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs]: Removed

**Review by Charles-Gagnon — APPROVED** [review]

**Review by alanrenmsft — APPROVED** [review]

**Pull request merged**